config/schema_helpers.go

```diff
@@ -60,27 +61,51 @@ func setupSchemaLoaders(schemaData string, schema *map[string]interface{}, schem
 	return nil
 }
 
+func oneOfValidTypes(oneOf interface{}, validTypes []string) []string {
+	validConditions := oneOf.([]interface{})
+
+	for _, validCondition := range validConditions {
+		condition := validCondition.(map[string]interface{})
+		if _, ok := condition["type"]; ok {
+			validTypes = append(validTypes, condition["type"].(string))
+		}
+	}
+
+	return validTypes
+}
+
 // gojsonschema doesn't provide a list of valid types for a property
 // This parses the schema manually to find all valid types
 func parseValidTypesFromSchema(schema map[string]interface{}, context string) []string {
 	contextSplit := strings.Split(context, ".")
 	key := contextSplit[len(contextSplit)-1]
 
-	definitions := schema["definitions"].(map[string]interface{})
-	service := definitions["service"].(map[string]interface{})
-	properties := service["properties"].(map[string]interface{})
-	property := properties[key].(map[string]interface{})
+	definitions := make(map[string]interface{})
+	if _, ok := schema["definitions"]; ok {
+		definitions = schema["definitions"].(map[string]interface{})
+	}
+
+	service := make(map[string]interface{})
+	if _, ok := definitions["service"]; ok {
+		service = definitions["service"].(map[string]interface{})
+	}
+
+	properties := make(map[string]interface{})
+	if _, ok := service["properties"]; ok {
+		properties = service["properties"].(map[string]interface{})
+	}
+
+	property := make(map[string]interface{})
+	if _, ok := properties[key]; ok {
+		property = properties[key].(map[string]interface{})
+	}
 
 	var validTypes []string
 
 	if val, ok := property["oneOf"]; ok {
-		validConditions := val.([]interface{})
-
-		for _, validCondition := range validConditions {
-			condition := validCondition.(map[string]interface{})
-			validTypes = append(validTypes, condition["type"].(string))
-		}
+		validTypes = oneOfValidTypes(val, validTypes)
 	} else if val, ok := property["$ref"]; ok {
+		logrus.Infof("REFERENCE %v\n", property["$ref"])
```

Comment:
I just removed it because it was a log that should have been left out of this PR.